tests/assembly/naked-fn-target-feature.rs

```diff
@@ -0,0 +1,173 @@
+
+#![crate_type = "lib"]
+#![feature(no_core, naked_functions, asm_experimental_arch)]
+#![feature(
+    avx512_target_feature,
+    s390x_target_feature,
+    powerpc_target_feature,
+    loongarch_target_feature,
+    m68k_target_feature
+)]
+#![no_core]
+
+extern crate minicore;
+use minicore::*;
+
+// x86_64-LABEL: vpclmulqdq:
+// x86_64: vpclmulqdq
+#[no_mangle]
+#[naked]
+#[cfg(target_arch = "x86_64")]
+#[target_feature(enable = "vpclmulqdq")]
+unsafe extern "C" fn vpclmulqdq() {
+    naked_asm!("vpclmulqdq zmm1, zmm2, zmm3, 4")
+}
+
+// aarch64-elf-LABEL: aes:
+// aarch64-elf: aese
+// aarch64-macho-LABEL: aes:
+// aarch64-macho: aese
+// aarch64-coff-LABEL: aes:
+// aarch64-coff: aese
+#[no_mangle]
+#[naked]
+#[cfg(target_arch = "aarch64")]
+#[target_feature(enable = "aes")]
+unsafe extern "C" fn aes() {
+    naked_asm!("aese.16b v0, v1")
+}
+
+// riscv64: sh1add:
+// riscv64: sh1add
+#[no_mangle]
+#[naked]
+#[cfg(target_arch = "riscv64")]
+#[target_feature(enable = "zba")]
+unsafe extern "C" fn sh1add() {
+    naked_asm!("sh1add a0, a1, a2", "ret");
+}
+
+#[cfg(target_arch = "s390x")]
+mod s390x {
+    use super::*;
+
+    // s390x: vector:
+    // s390x: vavglg
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector")]
+    unsafe extern "C" fn vector() {
+        naked_asm!("vavglg  %v0, %v0, %v0")
+    }
+
+    // s390x: vector_enhancements_1:
+    // s390x: vfcesbs
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector-enhancements-1")]
+    unsafe extern "C" fn vector_enhancements_1() {
+        naked_asm!("vfcesbs %v0, %v0, %v0")
+    }
+
+    // s390x: vector_enhancements_2:
+    // s390x: vclfp
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector-enhancements-2")]
+    unsafe extern "C" fn vector_enhancements_2() {
+        naked_asm!("vclfp   %v0, %v0, 0, 0, 0")
+    }
+
+    // s390x: vector_packed_decimal:
+    // s390x: vlrlr
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector-packed-decimal")]
+    unsafe extern "C" fn vector_packed_decimal() {
+        naked_asm!("vlrlr   %v24, %r3, 0(%r2)", "br      %r14")
+    }
+
+    // s390x: vector_packed_decimal_enhancement:
+    // s390x: vcvbg
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector-packed-decimal-enhancement")]
+    unsafe extern "C" fn vector_packed_decimal_enhancement() {
+        naked_asm!("vcvbg   %r0, %v0, 0, 1")
+    }
+
+    // s390x: vector_packed_decimal_enhancement_2:
+    // s390x: vupkzl
+    #[no_mangle]
+    #[naked]
+    #[target_feature(enable = "vector-packed-decimal-enhancement-2")]
+    unsafe extern "C" fn vector_packed_decimal_enhancement_2() {
+        naked_asm!("vupkzl  %v0, %v0, 0")
+    }
+}
+
+// powerpc64: altivec:
+// powerpc64: vsumsws
+#[no_mangle]
+#[naked]
+#[cfg(target_arch = "powerpc64")]
+#[target_feature(enable = "altivec")]
+unsafe extern "C" fn altivec() {
```

Comment:
On powerpc64-unknown-linux-gnu, altivec is enabled by default, so I think this #[target_feature] is no-op

```console
$ rustc --print cfg --target powerpc64-unknown-linux-gnu | grep target_feature
target_feature="altivec"
```

Reply:
right, I'm now using the `xxpermx` which is part of `power10-vector` https://godbolt.org/z/cbz1vqGsj